pkg/notes/client.go

```diff
@@ -48,21 +50,46 @@ type githubNotesClient struct {
 var _ Client = &githubNotesClient{}
 
 func (c *githubNotesClient) GetCommit(ctx context.Context, owner string, repo string, sha string) (*github.Commit, *github.Response, error) {
-	return c.ghc.Git.GetCommit(ctx, owner, repo, sha)
+	for retry := internal.GithubRetryer(internal.MaxGithubRetries, time.Sleep); ; {
```

Comment:
Can we avoid the for loop while introducing a `DefaultGithubRetryer` which automatically does the `for` and `time.Sleep`?

Reply:
I renamed things a bit, I think it makes them a bit clearer.

I also did what you suggested and added a `DefaultGithubErrChecker`.

> [...] which automatically does the `for` [...]

Because there are different types at play, I am not sure how to do that without punching a hole through the type system and type assert back and forth from / to `interface{}`
or such. If you have any suggestions on how to do that in a general way without doing the type assertions, let me know!